### ISSUE TITLE
Import Address or Transaction which was created using xpub of same wallet to get updated balance and use funds.

### ISSUE THREAD

**Issue by Adonis2115** [spesmilo/electrum#5779]
I have exported xpub and used it in one of my website to give a different address to all users and when funds are received on any address, it is not visible on electrum wallet. I can understand this as electrum can not watch infinite addresses. But it would be a great feature to import the address or transaction which was received (also in bulk) so it can be updated in the wallet and I can use funds from electrum itself.

PS: If it's possible now please let me know.

**Comment by SomberNight** [comment]
Is this a [gap limit](https://github.com/spesmilo/electrum-docs/blob/master/faq.rst#what-is-the-gap-limit) issue?
Or why do you think the transactions are not visible in Electrum?

**Comment by Adonis2115** [comment]
Yeah might be gap limit issue, the address which got generated is different than the 20 address provided by Electrum, while when address is generated using xpriv is the same as the 20 address in the order.

**Comment by ecdsa** [comment]
> the address which got generated is different than the 20 address provided by Electrum, while when address is generated using xpriv is the same as the 20 address in the order.

that does not sound like a gap limit issue...

**Comment by SomberNight** [comment]
@Adonis2115 I am not sure I understand.
- where did you get the xprv?
- do the addresses shown by electrum, and the addresses generated on your website, have the same script type? what is the first character of these addresses respectively?
- what plugin/etc are you using to generate addresses on the website?

> the address which got generated is different than the 20 address provided by Electrum, while when address is generated using xpriv is the same as the 20 address in the order

Could you please clarify that? Did you mean the following:
- Electrum with xprv (or seed?) generates addresses list A
- website plugin with xpub generates addresses list B
- website plugin with xprv generates addresses list B

**Comment by Adonis2115** [comment]
> where did you get the xprv?
I did not get it, I did create a wallet using mnemonic in electrum.

> do the addresses shown by electrum, and the addresses generated on your website, have the same script type? what is the first character of these addresses respectively?
I believe the script is same both address shown in electrum and generated using xpub programmatically start with letter 'm'

> what plugin/etc are you using to generate addresses on the website?
bitcore lib I am using to generate address using xpub.

> Electrum with xprv (or seed?) generates addresses list A
Electrum with mnemonic

> website plugin with xpub generates addresses list B
bitcore lib

> website plugin with xprv generates addresses list B
not generating using xprv

**Comment by ecdsa** [comment]
no idea how bitcore lib generates addresses from xpub. 
that is not an Electrum issue, though

**Comment by marksman0063** [comment]
Thanks, Dr David from HackAssets. Love the information and your working strategy. As a beginner, your tutorial guide at first seems a little bit confusing, but ultimately I learned a lot. Dr David, Manage investment project seems super interesting too. Invested 1Btc with him and his management, managed it to 7.6Btc in 3days. Thanks again Dave. I think if you trade crypto you should write on Hackassets via google mail, and google search about his management team
